lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnDriverRuntimeRestartManager.java

```diff
@@ -78,31 +78,44 @@ private YarnDriverRuntimeRestartManager(@Parameter(YarnEvaluatorPreserver.class)
   }
 
   /**
-   * Determines whether the application master has been restarted based on the container ID environment
+   * Determines the number of times the Driver has been submitted based on the container ID environment
    * variable provided by YARN. If that fails, determine whether the application master is a restart
-   * based on the number of previous containers reported by YARN.
-   * @return true if the application master is a restarted instance, false otherwise.
+   * based on the number of previous containers reported by YARN. In the failure scenario, returns 1 if restart, 0
+   * otherwise.
+   * @return > 0 if the application master is a restarted instance, 0 otherwise.
    */
   @Override
-  public boolean hasRestarted() {
+  public int getResubmissionAttempts() {
     final String containerIdString = getContainerIdString();
 
     if (containerIdString == null) {
       // container id should always be set in the env by the framework
       LOG.log(Level.WARNING, "Container ID is null, determining restart based on previous containers.");
-      return this.isRestartByPreviousContainers();
+      if (this.isRestartByPreviousContainers()) {
```

Comment:
You can consolidate those two pieces with condition of 
if (containerIdString == null || appAttemptID == null)